src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs

```diff
@@ -92,6 +97,32 @@ private static bool IsValidForTaskLikeTypeOnlyContext(ISymbol symbol, TSyntaxCon
             return symbol.IsAwaitableNonDynamic(context.SemanticModel, context.Position);
         }
 
+        private static bool IsValidForGenericConstraintContext(ISymbol symbol)
+        {
+            if (symbol.IsNamespace() ||
+                symbol.IsKind(SymbolKind.TypeParameter))
+            {
+                return true;
+            }
+
+            if (symbol is not INamedTypeSymbol namedType ||
+                symbol.IsDelegateType() ||
+                namedType.IsEnumType() ||
+                namedType.IsStructType())
```

Comment:
i think structs need to move into the list below. t hey can have nested types that would be valid for constraints.